cvat-ui/src/components/profile-page/profile-page.tsx

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { CombinedState } from 'reducers';
+
+import { UserOutlined, LockOutlined } from '@ant-design/icons';
+import { Col, Row } from 'antd/lib/grid';
+import Typography from 'antd/lib/typography';
+import Menu from 'antd/lib/menu';
+
+import { getTabFromHash } from 'utils/location-utils';
+import CVATLoadingSpinner from 'components/common/loading-spinner';
+import ProfileContent from './profile-content';
+import SecurityContent from './security-content';
+
+import './styles.scss';
+
+const { Title } = Typography;
+
+function ProfilePageComponent(): JSX.Element {
+    const { user, fetching, isPasswordChangeEnabled } = useSelector((state: CombinedState) => ({
+        user: state.auth.user,
+        fetching: state.auth.fetching,
+        isPasswordChangeEnabled: state.serverAPI.configuration.isPasswordChangeEnabled,
+    }));
+
+    const supportedTabs = isPasswordChangeEnabled ? ['profile', 'security'] : ['profile'];
+    const [activeTab, setActiveTab] = useState(getTabFromHash(supportedTabs));
+
+    const menuItems = [
+        {
+            key: 'profile',
+            icon: <UserOutlined />,
+            label: 'Profile',
+        },
+        ...(isPasswordChangeEnabled ? [{
+            key: 'security',
+            icon: <LockOutlined />,
+            label: 'Security',
+        }] : []),
+    ];
+
+    const onMenuClick = useCallback((key: string): void => {
+        setActiveTab(key);
+    }, []);
+
+    useEffect(() => {
+        const onHashChange = (): void => setActiveTab(getTabFromHash(supportedTabs));
+        window.addEventListener('hashchange', onHashChange);
+        return (): void => window.removeEventListener('hashchange', onHashChange);
+    }, [supportedTabs]);
```

Comment:
As supportedTabs is a new object each render, useEffect will be also called each render

Reply:
Removed this dependency.